### PR TITLE
357 keypoint detection demo

### LINKED ISSUE #357: Keypoint detection demo
Detect person keypoints and visualize them.

- [ ] Colored skeleton lines
- [ ] Python converter for model output
- [ ] Check if the model works correctly on batch larger than 1 and several streams.

### PR DESCRIPTION
**Pull request by dorgun**

### CODE REVIEW

File: samples/keypoint_detection/overlay.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
+from savant.deepstream.drawfunc import NvDsDrawFunc
+from savant.deepstream.meta.frame import NvDsFrameMeta
+from savant.utils.artist import Artist
+import numpy as np
```
Comment (abramov-oleg) [inline samples/keypoint_detection/overlay.py]: Need to run `make reformat`

File: samples/README.md [hunk 2]
```diff
@@ -77,6 +77,15 @@ Preview:
...
 ![](license_plate_recognition/assets/license-plate-recognition-400.webp)
 
+
+### Keypoint detection 
+
+Sample location: [keypoint_detection](./keypoint_detection)
+
+Preview:
+
+![](keypoint_detection/assets/shuffle_dance-400.webp)
```
Comment (abramov-oleg) [inline samples/README.md:87]: No assets in the PR.

File: samples/keypoint_detection/module.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
+name: ${oc.env:MODULE_NAME, 'keypoint_detection'}
```
Comment (abramov-oleg) [inline samples/keypoint_detection/module.yml]:
```suggestion
name: keypoint_detection
```

For consistency with incoming changes from Denis' PR.

File: savant/utils/artist/artist_gpumat.py [hunk 4]
```diff
@@ -248,6 +248,25 @@ def add_circle(
         self.__init_overlay()
         cv2.circle(self.overlay, center, radius, color, thickness, line_type)
 
+    def add_line(
+        self,
+        pt1: Tuple[int, int],
+        pt2: Tuple[int, int],
+        color: Tuple[int, int, int, int] = (255, 0, 0, 255),  # RGBA, Red,
+        thickness: int = 3,
+    ):
```
Comment (abramov-oleg) [inline savant/utils/artist/artist_gpumat.py]: Let's add a line type parameter with `cv2.LINE_AA` default value.

**Review by abramov-oleg — CHANGES_REQUESTED** [review]

File: samples/keypoint_detection/overlay.py (hunk 1 above)
Comment (dorgun) [reply samples/keypoint_detection/overlay.py]: formatted

File: samples/README.md (hunk 2 above)
Comment (dorgun) [reply samples/README.md:87]: Added

File: savant/utils/artist/artist_gpumat.py (hunk 4 above)
Comment (dorgun) [reply savant/utils/artist/artist_gpumat.py]: Added

**Review by abramov-oleg — APPROVED** [review]

**Pull request merged**